src/joinboost/preprocessor.py

```diff
@@ -1,20 +1,149 @@
-from joingraph import JoinGraph
+from typing import List
+from joinboost.joingraph import JoinGraph
+import json 
 
 
-class Preprocess:
-    def __init__(self, join_graph: JoinGraph):
-        self.join_graph = join_graph
-        # records the history of  preprocess
-        self.history = {}
-        self.view2table = {}  # keeps track of rename changes
+# Preprocess the join graph and store modifications.
+class Preprocessor:
+    def __init__(self):
+        self.view2table = {} # keeps track of rename changes
+        self.table2view = {}
+        self.jg = None
```

Comment:
i thought we just need some data structure:

self.rename={} that maps {table_name =>{old_col => new col}}